Rebrand exome-seek to xavier

samarth8392 opened the pull request:
**The branch was sourced from `main` but is being PR'ed on `dev`**

* Updated all instances of `exome-seek` with `xavier`
* Updated all instances of `Exome-seek` and/or `WES GATK4` with `XAVIER`
* Updated all instances of `module load singularity snakemake` with `module load ccbrpipeliner` 
* Removed all temp files created during rhel8 testing (e.g. `tools.rhel8.json`)
* Replaced all URLs in docs to the CCBR/XAVIER repo
* Updated all usage to reflect the new required `--runmode` command
* Added `bin` folder with `xavier`and redirect (like RENEE)
* Tested `./xavier` pipeline on biowulf (RHEL8)

kopardev commented on docs/usage/cache.md:
@samarth8392 ... get rid of the './' prefix in front of `xavier` in the code blocks. `module load ccbrpipeliner` is supposed to add `xavier` to `PATH`, so we do not need to be a specific folder to run `xavier`!

kopardev replied:
Also, you mentioned that the SIF cache folder was missing a couple of sif images... Is it because `xavier --sif-cache` did not pull them down? If that is the case then we need to fix the `--sif-cache` "option".

samarth8392 replied:
Sorry for the confusion but I was referring this issue for the `xavier_gui`. If we use the command 
` --sif-cache /data/CCBR_Pipeliner/SIFS/ ` like RENEE gui is supposed to [do](https://github.com/CCBR/guis/blob/fb893b53a0b44df140151a267d446a1dfc1a903e/bin/setup#L8C19-L8C44), then we get the following warning:
```
xavier (v1.1)
--Initializing
Warning: Local image "/data/CCBR_Pipeliner/SIFS/ccbr_wes_base_v0.1.0.sif" does not exist in singularity cache
Warning: Local image "/data/CCBR_Pipeliner/SIFS/ccbr_vcf2maf_v102.0.0.sif" does not exist in singularity cache
Warning: Local image "/data/CCBR_Pipeliner/SIFS/ccbr_mutect_v0.1.0.sif" does not exist in singularity cache
Warning: Local image "/data/CCBR_Pipeliner/SIFS/ccbr_multiqc_1.9_v0.0.1.sif" does not exist in singularity cache
```

and since `$ ./xavier cache --sif-cache xavier_cache` is not implemented, we cannot create cache. So what I wanted to say was if we are going to use SIF caches in CCBR pipeliner GUI, we need to add the aforementioned images.

kopardev commented on docs/usage/run.md:
```diff
@@ -54,12 +54,23 @@ Each of the following arguments are required. Failure to provide a required argu
+> This option defines the reference genome for your set of samples. On Biowulf, xavier does comes bundled with pre built reference files for human samples; however, it is worth noting that the pipeline does accept a pre-built resource bundle pulled with the cache sub command (coming soon). Currently, the pipeline only supports the human reference hg38; however, support for mouse reference mm10 wi ...
 >
 > ***Pre built Option***  
 > Here is a list of available pre built genomes on Biowulf: hg38.
```
@samarth8392 This "pre built genomes" is pretty small! .I think we need mm10 (and hg19) as well. Please, we also need to be able replace "on Biowulf" with "on Biowulf and FRCE". Do you have some rough documentation on how to create these "pre built genomes" support? May be create a set of script(s) to automatically handle this?? @dnousome  any thoughts?

kopardev commented on docs/usage/run.md:
```diff
@@ -90,15 +101,6 @@ Each of the following arguments are optional and do not need to be provided.
 > 
 > ***Example:*** `--help`
 
----  
```
@samarth8392  Is there any particular reason why `--dry-run` is removed from CLI? May be I confused you with my comments on Friday. I meant to remove dryrun from GUI, but keep it for now at CLI ... also will be replaced with `--stub` when we transition to Nextflow.

dnousome replied:
Dry run is still available (just moved to the `--runmode dryrun` flag suggested by @slsevilla)

kopardev commented on resources/runner:
```diff
@@ -237,7 +237,7 @@ set -euo pipefail
   --use-singularity --singularity-args "'-B $4'" --configfile="$3/config.json" \\
-  --printshellcmds --cluster-config "$3/config/cluster.json" \\
+  --printshellcmds --cluster-config "$3/cluster.json" \\
   --cluster "${CLUSTER_OPTS}" --keep-going --restart-times 3 -j 500 \\
```
Can you change CLUSTER_OPTS like [this](https://github.com/CCBR/RENEE/blob/474884f5755316bc5da5f8ce83fce85f472bdf2f/resources/runner#L163) and later add `--gres` like [this](https://github.com/CCBR/RENEE/blob/474884f5755316bc5da5f8ce83fce85f472bdf2f/resources/runner#L180) .. this is required as gres is not available on FRCE. Also compare other things with the v2.2 RENEE runner

samarth8392 replied:
both `runner` and `config.cluster.json` has been updated.

kopardev commented on src/run.py:
```diff
@@ -180,11 +180,16 @@ def setup(sub_args, repo_path, output_path, create_nidap_folder_YN = 'no',links=
     hostname = os.getenv("HOSTNAME")
     if hostname == "fsitgl-head01p.ncifcrf.gov":
         shorthostname = "frce"
+    elif hostname == "biowulf8.nih.gov":
```
remove rhel8 or biowulf8 stuff .. that is not required after the migration

kopardev commented on xavier:
```diff
@@ -342,33 +344,32 @@ def parsed_arguments():
 
           # Step 2A.) Initialize the pipeline
-          ./exome-seek run \\
+          ./xavier run \\
```
get rid of the `./`

kopardev commented on xavier:
```diff
@@ -342,33 +344,32 @@ def parsed_arguments():
           
           # Step 2B.) Dry-run the pipeline
-          ./exome-seek run \\
+          ./xavier run \\
```
no `./`

kopardev commented on xavier:
```diff
@@ -342,33 +344,32 @@ def parsed_arguments():
-          # It is recommended running exome-seek in this mode.
-          ./exome-seek run \\
+          # It is recommended running xavier in this mode.
+          ./xavier run \\
```
no `./`

kopardev commented:
Add:

- [jobby](https://github.com/CCBR/RENEE/blob/v2.2/resources/jobby)
- [run_jobby_on_snakemake_log](https://github.com/CCBR/RENEE/blob/v2.2/resources/run_jobby_on_snakemake_log)
- [spooker](https://github.com/CCBR/RENEE/blob/v2.2/resources/spooker) 

to the `resources` folder. Then,
add `onerror` and `onsuccess` blocks like [this](https://github.com/CCBR/RENEE/blob/474884f5755316bc5da5f8ce83fce85f472bdf2f/workflow/Snakefile#L208)

kopardev commented:
Change `cluster.json`'s `default` like [this](https://github.com/CCBR/XAVIER/blob/f098e3642f4428b8174ba6ffa55a9821f9c4446f/config/cluster.biowulf.json#L2). This will allow us to capture rulename, samplename and other metadata in the filename of the log/err file itself. This is also liked with the changes that I proposed in the `runner` script.

samarth8392 commented on config/templates/tools.json:
We only use Dockers but these jsons are just templates in case anyone wants to run on biowulf without the images. I updated this according to Issue #15 during rhel8 testing and updated to have only the latest versions available. However, testing  to see if the pipeline runs without docker images and only using the environment modules within the biowulf8 failed. see details [here](https://github.com/CCBR/techdev_efforts/issues/20#issuecomment-1589657509).
`tools.json` only gets appended but never used. See #19

samarth8392 commented:
I have addressed all your comments and have updated all the files as necessary. see commit be88f577fba5585cfc262f39c4a9af6436f920a9.
However, I only copied `resources/spooker` and `jobby` files from RENEE and @dnousome  needs to update them before the merge.

kopardev reviewed (approved):
Looks good!
